pylint/checkers/classes.py

```diff
@@ -960,6 +960,22 @@ def _check_unused_private_attributes(self, node: astroid.ClassDef) -> None:
         for assign_attr in node.nodes_of_class(astroid.AssignAttr):
             if not is_attr_private(assign_attr.attrname):
                 continue
+
+            # Logic for checking false positive when using __new__,
+            # Get the returned object names of the __new__ magic function
+            # Then check if the attribute was consumed in other instance methods
+            acceptable_obj_names = ["self"]
+            if (
+                isinstance(assign_attr.scope(), astroid.FunctionDef)
```

Comment:
You are using `assign_attr.scope()` multiple times here. As this is a possibly recursive function call, it might be better to do it just once and assign the result to a variable (e.g., `scope`). That will also make it easier for type checkers.